src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -182,6 +185,33 @@ export async function activate(vscodeType: typeof vscodeapi, context: ExtensionC
             localRegistrations.length = 0;
         });
 
+        const provider = new BlazorDebugConfigurationProvider(logger, vscodeType);
+        context.subscriptions.push(vscodeType.debug.registerDebugConfigurationProvider('blazor', provider));
+
+        /**
+         * On non-Windows platforms, we need to terminate the Blazor
+         * dev server and its child processes.
+         */
+        vscodeType.debug.onDidTerminateDebugSession(event => {
+            logger.logVerbose('Terminating debugging session...');
+            if (process.platform !== 'win32') {
+                psList().then(processes => {
+                    const devserver = processes.find(
+                        (process: psList.ProcessDescriptor) => !!(process && process.cmd && process.cmd.includes('blazor-devserver')),
```

Comment:
Just learnt why `!!` is useful. Very interesting.

Reply:
TypeScript generates a union type when the && operator is used so the type for the `process && process.cmd && process.cmd.includes('blazor-devserver')` clause is `undefined | null | string` because the first value in the clause `process` can be any one of these. I used the `!!` operator to coerce the type to a `boolean` which is the type signature that the `find` function expects (e.g. `(process: ProcessDescriptor) => boolean`)